python/pyproject.toml

```diff
@@ -34,7 +34,7 @@ dependencies = [
 
 [project.optional-dependencies]
 docs = ["markdown", "pyyaml", "sphinx", "sphinx_rtd_theme"]
-lint = ["ruff"]
+lint = ["ruff>=0.2.0"]
```

Comment:
Is this pin required for ruff to run?

Reply:
According to the release notes I've linked above, yes:
> Ruff plugin
>
> Many of the changes covered in the 2.0 release notes and in this migration guide can be automatically adapted > in downstream code with a dedicated [Ruff](https://docs.astral.sh/ruff/) rule, namely rule [NPY201](https://docs.astral.sh/ruff/rules/numpy2-deprecation/).
> You should install ruff>=0.2.0 and add the NPY201 rule to your pyproject.toml: